llvm/lib/Support/APFloat.cpp

```diff
@@ -1165,7 +1200,7 @@ IEEEFloat::IEEEFloat(const fltSemantics &ourSemantics, integerPart value) {
 
 IEEEFloat::IEEEFloat(const fltSemantics &ourSemantics) {
   initialize(&ourSemantics);
-  makeZero(false);
+  ourSemantics.hasZero ? makeZero(false) : makeSmallestNormalized(false);
```

Comment:
It's not obvious to me that makeSmallestNormalized is the alternative. What's wrong with makeZero? You'll get a zero exponent, and an empty significand?

Reply:
The Float8E8M0FNU format does not have a representation for zero. So, we are initializing with the closest possible representation (which happens to be the smallestNormalized for this format).

Zero is special with its own 'fcZero' category. Since we have a normal value here, we cannot use makeZero() to initialize it. (i.e. we need this to be categorized as 'fcNormal' and not as 'fcZero').

I have added a comment regarding this in the latest revision.